omod/src/main/java/org/openmrs/module/idgen/rest/search/LogEntrySearchHandler.java

```diff
@@ -0,0 +1,77 @@
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.module.idgen.rest.search;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Date;
+import java.util.List;
+import org.openmrs.module.idgen.IdentifierSource;
+import org.openmrs.module.idgen.LogEntry;
+import org.openmrs.module.idgen.service.IdentifierSourceService;
+import org.openmrs.module.idgen.web.controller.IdgenRestController;
+import org.openmrs.module.webservices.rest.web.ConversionUtil;
+import org.openmrs.module.webservices.rest.web.RequestContext;
+import org.openmrs.module.webservices.rest.web.RestConstants;
+import org.openmrs.module.webservices.rest.web.resource.api.PageableResult;
+import org.openmrs.module.webservices.rest.web.resource.api.SearchConfig;
+import org.openmrs.module.webservices.rest.web.resource.api.SearchHandler;
+import org.openmrs.User;
+import org.openmrs.api.UserService;
+import org.openmrs.api.context.Context;
+import org.openmrs.module.webservices.rest.web.resource.api.SearchQuery;
+import org.openmrs.module.webservices.rest.web.resource.impl.EmptySearchResult;
+import org.openmrs.module.webservices.rest.web.resource.impl.NeedsPaging;
+import org.openmrs.module.webservices.rest.web.response.ResponseException;
+import org.springframework.stereotype.Component;
+
+@Component
+public class LogEntrySearchHandler implements SearchHandler {
+
+    private final SearchConfig searchConfig = new SearchConfig("default", RestConstants.VERSION_1 + IdgenRestController.IDGEN_NAMESPACE + "/logentry",
+            Arrays.asList("1.8.*", "1.9.*", "1.10.*", "1.11.*", "1.12.*", "2.0.*", "2.1.*"),
+            Arrays.asList(new SearchQuery.Builder(
+                    "Allows you to find log of ID Generation Activities by Source Name, Identifier contents,Generated Date Range,Comment contents and User who generated the log entry")
+                            .withOptionalParameters("source", "identifier", "fromDate", "toDate", "comment",
+                                    "generatedBy")
+                            .build()));
+
+    @Override
+    public PageableResult search(RequestContext context) throws ResponseException {
+        IdentifierSourceService identifierSourceService = Context.getService(IdentifierSourceService.class);
+        UserService service = Context.getUserService();
+
+        String source = context.getRequest().getParameter("source");
+        String fromDate = context.getRequest().getParameter("fromDate");
+        String toDate = context.getRequest().getParameter("toDate");
+        String identifier = context.getRequest().getParameter("identifier");
+        String comment = context.getRequest().getParameter("comment");
+        String generatedBy = context.getRequest().getParameter("generatedBy");
+
+        List<LogEntry> logEntries = new ArrayList<LogEntry>();
```

Comment:
Yo do not need the above declaration. You can just do it at the point when you are assigning it the actual value.

Reply:
Fixed